wrappers/src/table_cs.cpp

```diff
@@ -66,33 +60,33 @@ REALM_EXPORT Row* table_get_link(Table* table_ptr, size_t column_ndx, size_t row
   });
 }
 
-REALM_EXPORT SharedLinkViewRef* table_get_linklist(Table* table_ptr, size_t column_ndx, size_t row_ndx)
+REALM_EXPORT SharedLinkViewRef* table_get_linklist(Table* table_ptr, size_t column_ndx, size_t row_ndx, NativeException::Marshallable& ex)
 {
-    return handle_errors([&]() -> SharedLinkViewRef* {
-        SharedLinkViewRef sr = std::make_shared<LinkViewRef>(table_ptr->get_linklist(column_ndx, row_ndx));
-        return new SharedLinkViewRef{ sr };  // weird double-layering necessary to get a raw pointer to a shared_ptr
-    });
+  return handle_errors(ex, [&]() -> SharedLinkViewRef* {
+    SharedLinkViewRef sr = std::make_shared<LinkViewRef>(table_ptr->get_linklist(column_ndx, row_ndx));
+    return new SharedLinkViewRef{ sr };  // weird double-layering necessary to get a raw pointer to a shared_ptr
```

Comment:
Moving `sr` into the `SharedLinkViewRef` constructor will avoid an unnecessary increment / decrement of the `shared_ptr`'s refcount.